app/app.go

```diff
@@ -840,52 +840,15 @@ func NewApplication(
 
 			//add more upgrade instructions
 			ctx.Logger().Info("Running revert of tombstoning")
-			err := upgrades.RevertCosTombstoning(ctx, app.SlashingKeeper, app.MintKeeper, &bankkeeper.BaseKeeper{}, app.StakingKeeper)
+			err := upgrades.RevertCosTombstoning(ctx, app.SlashingKeeper, app.MintKeeper, app.BankKeeper, app.StakingKeeper)
 			if err != nil {
 				panic(fmt.Sprintf("failed to revert tombstoning: %s", err))
 			}
 
-			// RunMigrations twice is just a way to make auth module's migrates after staking
-			// send stk/uatom to persistence1zl42hd5h9c7z4ej43fhss9nvgm6nuad0js8z6n (for https://www.mintscan.io/cosmos/txs/DE691EC8EBB5A79E2AB421291660111E893823CA0CC9EBDED5E3C72B503067C3 sending amount to reward address)
-			// at c_value 0.999142233051758540 = 44961400stk/uatom (`curl -X GET -H "Content-Type: application/json" -H "x-cosmos-block-height: 8650000" 'https://rest.core.persistence.one/pstake/lscosmos/v1beta1/c_value'`)
-			PstakeUpgradeV6 := func(ctx sdk.Context, k lscosmoskeeper.Keeper) {
-				atomTVU := k.GetDepositAccountAmount(ctx).
-					Add(k.GetIBCTransferTransientAmount(ctx)).
-					Add(k.GetDelegationTransientAmount(ctx)).
-					Add(k.GetStakedAmount(ctx)).
-					Add(k.GetHostDelegationAccountAmount(ctx))
-
-				mintedAmount := k.GetMintedAmount(ctx)
-				mintDenom := k.GetHostChainParams(ctx).MintDenom
-				if atomTVU.LTE(mintedAmount) {
-					return
-				}
-				toNewMint := atomTVU.Sub(mintedAmount)
-				if ctx.ChainID() == "core-1" {
-					if toNewMint.GT(sdk.NewInt(44961400)) {
-						mischiefUserAddress := sdk.MustAccAddressFromBech32("persistence1zl42hd5h9c7z4ej43fhss9nvgm6nuad0js8z6n")
-						toSendUser := sdk.NewInt(44961400)
-						err := k.MintTokens(ctx, sdk.NewCoin(mintDenom, toSendUser), mischiefUserAddress)
-						if err != nil {
-							k.Logger(ctx).Error("Failed to mint and send 44961400stk/uatom to persistence1zl42hd5h9c7z4ej43fhss9nvgm6nuad0js8z6n")
-						}
-						pstakeFeeAddress := sdk.MustAccAddressFromBech32(k.GetHostChainParams(ctx).PstakeParams.PstakeFeeAddress)
-						remainingAmount := toNewMint.Sub(toSendUser)
-						err = k.MintTokens(ctx, sdk.NewCoin(mintDenom, remainingAmount), pstakeFeeAddress)
-						if err != nil {
-							k.Logger(ctx).Error("Failed to mint and send remainingAmount to pstakeFeeAddress")
-						}
-					}
-				}
-				if ctx.ChainID() == "test-core-1" {
-					pstakeFeeAddress := sdk.MustAccAddressFromBech32(k.GetHostChainParams(ctx).PstakeParams.PstakeFeeAddress)
-					err := k.MintTokens(ctx, sdk.NewCoin(mintDenom, toNewMint), pstakeFeeAddress)
-					if err != nil {
-						k.Logger(ctx).Error("Failed to mint and send toNewMint to pstakeFeeAddress")
-					}
-				}
+			err = upgrades.MintPstakeTokens(ctx, app.LSCosmosKeeper)
+			if err != nil {
+				panic(fmt.Sprintf("failed to mint pstake tokens: %s", err))
 			}
-			PstakeUpgradeV6(ctx, *app.LSCosmosKeeper)
```

Comment:
@puneet2019 i changed this to a pointer since now all the keepers are basically pointers

Reply:
looks good, panic error or log error??

Reply:
up to you. Should we make log for both pstake minting and this pstake minting?
My considerations would be, we should panic so that if the upgrade handler does not run, we would be able to fix the upgrade at that time itself, instead of having to go though the whole upgrade process again.